modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/generalized_plane_strain.md

```diff
@@ -0,0 +1,125 @@
+\end{matrix}
+\end{equation}
+in $A$ where $\left\{ \bar{f} \right\}$ are the body forces.
+
+The constitutive equations in terms of stress-strain relationship are given as
+\begin{equation}
+\left\{ \begin{matrix}
+\sigma_{xx} \\
+\sigma_{yy} \\
+\sigma_{zz} \\
+\tau_{xy}
+\end{matrix} \right\}
+=
+\left[ \begin{matrix}
+C_{11} & C_{12} & C_{13} & 0 \\
+C_{12} & C_{22} & C_{23} & 0 \\
+C_{13} & C_{23} & C_{33} & 0 \\
+0 & 0 & 0 & C_{66}
+\end{matrix} \right]
+\left( \left\{ \begin{matrix}
+\epsilon_{xx} \\
+\epsilon_{yy} \\
+\epsilon_{zz} \\
+\gamma_{xy}
+\end{matrix} \right\}
+-
+\left\{ \begin{matrix}
+\alpha_{x} \\
+\alpha_{y} \\
+\alpha_{z} \\
+0
+\end{matrix} \right\} \Delta T \right)
+\end{equation}
+where $C_{ij}$ are material's stiffness coefficients using Voigt notation, $\alpha_{x}$, $\alpha_{y}$ and $\alpha_{z}$ are the material's thermal expansion coefficients and $\Delta T$ is the change in temperature.
+
+A further condition is required associated with the out-of-plane direction. If a constant strain is prescribed as the deformation compatibility condition in the $z$-direction
+\begin{equation}
+\epsilon_{zz} = \bar{\epsilon}_{zz}
+\end{equation}
+Alternatively, a force as the stress resultant $\bar{N}_{z}$ in the $z$-direction can be prescribed, the condition is the equilibrium condition in $z$-direction, given as
+\begin{equation}
+N_{z} = \int_{A}{\sigma_{zz}dA} = \bar{N}_{z}
+\end{equation}
+The stress resultant $N_{z}$ conjugates with constant strain $\epsilon_{zz}$.
+
+The formulation above for generalized plane strain problem shares many similarities with that of the conventional plane stress or plane strain problem. All the differences are associated with the introduction of an additional degree of freedom for the out-of-plane direction.
+
+## Implementation and Usage
+
+The out-of-plane strain is a scalar variable, and it can be added to the standard system of equations for a mechanics problem, where $\bf{u}_x$ and $\bf{u}_y$ represent
+the displacement vectors in the $x$ and $y$ directions, $\bf{f}_x$ and $\bf{f}_y$ represent the corresponding reaction forces. The partitioned linearized system of equations, in which the block entries in the stiffness matrix are represented by subscripted $\bf{K}$ terms, can be written including the scalar strain variable as follows:
+\begin{equation}
+\left[
+\begin{array}{cc|c}
+\bf{K}_{xx} & \bf{K}_{xy} & \vdots \\
+\bf{K}_{yx} & \bf{K}_{yy} & \vdots \\ \hline
+\cdots & \cdots & K_{zz}
+\end{array}
+\right]
+\left\{
+\begin{array}{c}
+\bf{u}_x\\
+\bf{u}_y\\ \hline
+\epsilon_{zz}
+\end{array}
+\right\}
+=
+\left\{
+\begin{array}{c}
+\bf{f}_x\\
+\bf{f}_y\\ \hline
+N_{z}
+\end{array}
+\right\}
+\end{equation}
+
+The off-diagonal entries are nonzero, but not shown here.
+
+### List of MOOSE objects used in the implementation
+
+Objects available for generalized plane strain:
+- [Stress Divergence Kernel](/StressDivergence.md): in-plane equilibrium equation
```

Comment:
Add a new line before the start of all the lists to get the lists to render properly.

A table might also be a nice way to present all of these different classes, ect, see the docs for CavityPressureAction